completion_aggregator/tasks/aggregation_tasks.py

```diff
@@ -201,18 +220,47 @@ def update_for_aggregator(self, block, affected_aggregators, force):
             if modified is not None:
                 last_modified = max(last_modified, modified)
         if self._aggregator_needs_update(block, last_modified, force):
-            obj, _ = Aggregator.objects.submit_completion(
-                user=self.user,
-                course_key=self.course_key,
-                block_key=block,
-                aggregation_name=block.block_type,
-                earned=total_earned,
-                possible=total_possible,
-                last_modified=last_modified,
-            )
-            self.aggregators[obj.block_key] = obj
+
+            self._is_aggregator_data_valid(self.user, self.course_key, block)
+            last_modified_unaware = make_datetime_timezone_unaware(last_modified)
+            created_at_unaware = make_datetime_timezone_unaware(timezone.now())
+            updated_at_unaware = make_datetime_timezone_unaware(timezone.now())
+            percent = get_percent(total_earned, total_possible)
+            self.processed_list.append({
+                'user': self.user,
+                'course_key': self.course_key,
+                'block_key': block,
+                'aggregation_name': block.block_type
+            })
+            # Param order: (user_id, course_key, block_key, block_type, earned, possible, percent,
+            #               last_modified, created, updated)
+            self.aggregator_list.append([
+                self.user.id,
+                self.course_key,
+                block,
+                block.block_type,
+                total_earned,
+                total_possible,
+                percent,
+                last_modified_unaware,
+                created_at_unaware,
+                updated_at_unaware
+            ])
         return CompletionStats(earned=total_earned, possible=total_possible, last_modified=last_modified)
 
+    def update_aggregator_at_once(self):
+        """
+        Update the collection of aggregator object using mysql insert on duplicate update query
+
+        After update, store the reference of obj in `aggregators` map
+        """
+        if self.aggregator_list:
+            with connection.cursor() as cur:
+                cur.executemany(INSERT_ON_DUPLICATE_UPDATE_QUERY, self.aggregator_list)
+            for qparams in self.processed_list:
+                obj = Aggregator.objects.get(**qparams)
+                self.aggregators[obj.block_key] = obj
```

Comment:
You are right, This will be N get queries. 
So I have stored a map of `block_key : { 'last_modified': 'last_modified_value'}`, because all we are doing is accessing the value of last modified from this.